splinterd/src/daemon.rs

```diff
@@ -1458,13 +1456,61 @@ fn create_registry(
     (unified_registry, registry_shutdown_handle)
 }
 
-fn parse_registry_arg(registry: &str) -> Result<(&str, &str), &str> {
+// Parses a registry argument, returning the uri scheme (defaulting to file) and remaining uri data
+fn parse_registry_arg(registry: &str) -> (&str, &str) {
     let mut iter = registry.splitn(2, "://");
-    let scheme = iter
+    let first = iter
         .next()
         .expect("str::split cannot return an empty iterator");
-    let path = iter.next().ok_or("No URI scheme provided")?;
-    Ok((scheme, path))
+    match iter.next() {
```

Comment:
Non-blocking: Since you are updating this, it might make sense to remove the expect and follow this style: https://github.com/Cargill/splinter/blob/main/libsplinter/src/rest_api/auth/mod.rs#L177-L183  adding a case of 
```rust
    (None, None) => unreachable!()
```
makes it much more clear then the above expect.

Reply:
Done :+1: